public/static/js/task.js

```diff
@@ -69,575 +69,595 @@ mciModule.controller('TaskHistoryDrawerCtrl', function($scope, $window, $locatio
+        $location.path('log/' + currentLogs);
+      };
+
+      $scope.formatTimestamp = function(logEntry, minVersion) {
+        if (!logEntry.version || logEntry.version < minVersion) {
+          return '';
         }
-      });
-    }
-  }
-});
-
-mciModule.controller('TaskLogCtrl', ['$scope', '$timeout', '$http', '$location', '$window', '$filter', 'notificationService', function($scope, $timeout, $http, $location, $window, $filter, notifier) {
-  $scope.logs = 'Loading...';
-  $scope.task = {};
-  $scope.eventLogs = 'EV';
-  $scope.systemLogs = 'S';
-  $scope.agentLogs = 'E';
-  $scope.taskLogs = 'T';
-  $scope.allLogs = 'ALL';
-  $scope.userTz = $window.userTz;
-
-  var logSpec = $location.path().split('/');
-  $scope.currentLogs = logSpec[2] || $scope.taskLogs;
-
-  $scope.$watch('currentLogs', function() {
-    $scope.getLogs();
-  });
-
-  $scope.setCurrentLogs = function(currentLogs) {
-    $scope.logs = 'Loading...';
-    $scope.currentLogs = currentLogs;
-    $location.path('log/' + currentLogs);
-  };
-
-  $scope.formatTimestamp = function(logEntry, minVersion) {
-    if (!logEntry.version || logEntry.version < minVersion) {
-      return '';
-    }
 
-    var converter = $filter('convertDateToUserTimezone');
-    var format = 'YYYY/MM/DD HH:mm:ss.SSS';
-    var timestamp = converter(logEntry.timestamp, $scope.userTz, format);
-    return '[' + timestamp + '] '
-  }
+        var converter = $filter('convertDateToUserTimezone');
+        var format = 'YYYY/MM/DD HH:mm:ss.SSS';
+        var timestamp = converter(logEntry.timestamp, $scope.userTz, format);
+        return '[' + timestamp + '] '
+      }
 
-  $scope.getLogs = function() {
-    $http.get('/json/task_log/' + $scope.taskId + '/' + $scope.task.execution + '?type=' + $scope.currentLogs).then(
-    function(resp) {
-      var data = resp.data;
-      if ($scope.currentLogs == $scope.eventLogs) {
-        $scope.eventLogData = data.reverse()
-      } else {
-        if (data && data.LogMessages) {
-          //read the log messages out, and reverse their order (since they are returned backwards)
-          $scope.logs = _.map(data.LogMessages, function(entry) {
-            var msg = entry.m.replace(/&#34;/g, '"');
-            var date = new Date(entry.ts);
-            return {
-              message: msg,
-              severity: entry.s,
-              timestamp: date,
-              version: entry.v
-            };
+      $scope.getLogs = function() {
+        $http.get('/json/task_log/' + $scope.taskId + '/' + $scope.task.execution + '?type=' + $scope.currentLogs).then(
+          function(resp) {
+            var taskScheduledStatus = "TASK_SCHEDULED";
+            var data = resp.data;
+            if ($scope.currentLogs == $scope.eventLogs) {
+              // remove any scheduled events other than the most recent one
+              var tmp = data.reverse();
+              var foundScheduled = false;
+              for (var i = tmp.length-1; i >=0; i--) {
```

Comment:
You reverse `tmp`, and then iterate over it in reverse order.

Reply:
It's a bit confusing but it's necessary. The data itself needs to be reversed to display in chronological order. I need to iterate through it in reverse because I'm deleting elements from it while iterating